spec/code-snippets/DeclarationReferences.ts

```diff
@@ -0,0 +1,443 @@
+}
+
+//---------------------------------------------------------
+// Merged declarations with extension of the same thing
+
+/**
+ * Shortest name:  {@link Merged11[interface]}
+ * Full name:      {@link Merged11[interface]}
+ */
+export interface Merged11 {
+  /**
+   * Shortest name:  {@link Merged11[interface].x}
+   * Full name:      {@link Merged11[interface].x}
+   */
+  x: string;
+}
+
+// (MUST NOT have TSDoc, because this augments an already documented interface)
+export interface Merged11 {
+  /**
+   * Shortest name:  {@link Merged11[interface].y}
+   * Full name:      {@link Merged11[interface].y}
+   */
+  y: string;
+}
+
+/**
+ * Shortest name:  {@link Merged11[namespace]}
+ * Full name:      {@link Merged11[namespace]}
+ */
+export namespace Merged11 {
+  /**
+   * Shortest name:  {@link Merged11[namespace].x}
+   * Full name:      {@link Merged11[namespace].x}
+   */
+  export let x: string = '';
+}
+
+// (MUST NOT have TSDoc, because this augments an already documented interface)
+export namespace Merged11 {
+  /**
+   * Shortest name:  {@link Merged11[namespace].y}
+   * Full name:      {@link Merged11[namespace].y}
+   */
+  export let y: string = '';
+}
+
+
+//---------------------------------------------------------
+// Enum members
+
+/**
+ * Shortest name:  {@link Enum12}
+ * Full name:      {@link Enum12[enum]}
+ */
+export const enum Enum12 {
+  /**
+   * Shortest name:  {@link Enum12.member13}
+   * Full name:      {@link Enum12[enum].member13}
+   */
+  member13
+}
+
+// (MUST NOT have TSDoc, because this augments an already documented enum)
+export const enum Enum12 {
+  /**
+   * Shortest name:  {@link Enum12.member14}
+   * Full name:      {@link Enum12[enum].member14}
+   */
+  member14 = 14
+}
+
+
+//---------------------------------------------------------
+// Malformed names
+
+/**
+ * Shortest name:  {@link Class15}
+ * Full name:      {@link Class15[class]}
+ */
+export class Class15 {
+  /**
+   * Shortest name:  {@link Class15."abc. def"}
+   * Full name:      {@link Class15[class]."abc. def"[static]}
+   */
+  public static 'abc. def': string = 'static member with malformed characters';
+
+  /**
+   * Shortest name:  {@link Class15."abc. def"}
+   * Full name:      {@link Class15[class]."abc. def"[instance]}
+   */
+  public 'abc. def': string = 'instance member with malformed characters';
+
+  /**
+   * Shortest name:  {@link Class15.static}
+   * Also valid:     {@link Class15."static"}
+   * Full name:      {@link Class15[class].static[static]}
+   */
+  public static static: string = 'static member using keyword as name';
+
+  /**
+   * Shortest name:  {@link Class15."\uD842\uDFB7"}
+   * Full name:      {@link Class15[class]."\uD842\uDFB7"[instance]}
+   *
+   * NOTE: The string in double quotes is parsed using JSON.parse(), which converts
+   * this surrogate pair expression to the corresponding Unicode character.
+   */
+  public '𠮷': string = 'instance member using JSON unicode escapes';
```

Comment:
>string [](start = 15, length = 6)

It might be useful to include an example of a member containing an allowed non-ASCII character (`ß`)